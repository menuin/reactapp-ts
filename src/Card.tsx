import React from "react";

const Card = ({
  number,
  cvc,
  date,
}: {
  number: number;
  cvc: string;
  date: string;
}) => {
  return (
    <div>
      number:{number}
      cvc:{cvc}
      date:{date}
    </div>
  );
};

export default Card;
