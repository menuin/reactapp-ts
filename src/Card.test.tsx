import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

describe("Card Template test", () => {
  const component = render(
    <Card number={1234123412341234} cvc={"012"} date={"08/24"}></Card>
  );

  expect(component.container).toMatchSnapshot();

  it("number", () => {
    expect(component.getByText(/number:(\d{16})/g).innerHTML).toMatch(/\d{16}/);
    expect(component.getByText(/cvc:\d{3}/g).innerHTML).toMatch(/\d{3}/g);
  });
});
