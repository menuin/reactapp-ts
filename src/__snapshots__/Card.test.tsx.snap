// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div>
    number:
    1234123412341234
    cvc:
    012
    date:
    08/24
  </div>
</div>
`;
